framework/src/meshgenerators/XYDelaunayGenerator.C

```diff
@@ -101,9 +128,21 @@ XYDelaunayGenerator::XYDelaunayGenerator(const InputParameters & parameters)
     _refine_holes(getParam<std::vector<bool>>("refine_holes")),
     _desired_area(getParam<Real>("desired_area")),
     _desired_area_func(getParam<std::string>("desired_area_func")),
+    _use_auto_area_func(getParam<bool>("use_auto_area_func")),
+    _auto_area_func_default_size(getParam<Real>("auto_area_func_default_size")),
+    _auto_area_func_default_size_dist(getParam<Real>("auto_area_func_default_size_dist")),
+    _auto_area_function_num_points(getParam<unsigned int>("auto_area_function_num_points")),
+    _auto_area_function_power(getParam<Real>("auto_area_function_power")),
     _algorithm(parameters.get<MooseEnum>("algorithm")),
     _verbose_stitching(parameters.get<bool>("verbose_stitching"))
 {
+  if ((_desired_area > 0.0 && !_desired_area_func.empty()) ||
+      (_desired_area > 0.0 && _use_auto_area_func) ||
+      (!_desired_area_func.empty() && _use_auto_area_func))
```

Comment:
We ll want to check for all the automatic area parameters here instead. Otherwise they can be specified by the user then ignored because another method is in use